service/history/timerGate.go

```diff
@@ -117,6 +148,65 @@ func (timerGate *TimerGateImpl) Update(nextTime time.Time) bool {
 }
 
 // Close shutdown the timer
-func (timerGate *TimerGateImpl) Close() {
+func (timerGate *LocalTimerGateImpl) Close() {
 	close(timerGate.closeChan)
 }
+
+// NewRemoteTimerGate create a new timer gate instance
+func NewRemoteTimerGate() RemoteTimerGate {
```

Comment:
Lot of implementation between Local and Remote looks the same.  Can we refactor it in a way to reuse code?

I'm OK with copy/pasting code if this is temporary.  But make sure create an issue to revisit this later.

Reply:
this is actually not a copy paste, although in general they looks the same, however they are backed by different thing, one with go's timer, one with comparison of new time.